qiskit_nature/second_q/mappers/tapered_qubit_mapper.py

```diff
@@ -0,0 +1,197 @@
+
+from .qubit_mapper import QubitMapper, _ListOrDict
+
+if TYPE_CHECKING:
+    from qiskit_nature.second_q.problems import ElectronicStructureProblem
+
+logger = logging.getLogger(__name__)
+
+
+class TaperedQubitMapper(QubitMapper):  # pylint: disable=missing-class-docstring
+    """The wrapper around qubit mappers implementing the logic to reduce the size of a problem (operator)
+    based on mathematical `Z2Symmetries` that can be automatically detected in the operator.
+
+    The following attributes can be read and updated once the ``TaperedQubitMapper`` object has been
+    constructed.
+
+    Attributes:
+        mapper: Object defining the mapping of second quantized operators to Pauli operators.
+        z2symmetries: Symmetries to use to reduce the Pauli operators.
+    """
+
+    def __init__(
+        self,
+        mapper: QubitMapper,
+        z2symmetries: Z2Symmetries = Z2Symmetries([], [], []),
+    ):
+        """
+
+        Args:
+            mapper: ``QubitMapper`` object implementing the mapping of second quantized operators to
+                Pauli operators.
+            z2symmetries: ``Z2Symmetries`` object defining the symmetries to use to reduce the Pauli
+                operators.
+        """
+        super().__init__()
+        self.mapper: QubitMapper = mapper
+        self.z2symmetries = z2symmetries
+
+    @classmethod
+    def from_problem(
+        cls, mapper: QubitMapper, problem: "ElectronicStructureProblem"
+    ) -> "TaperedQubitMapper":
+        # implies the previous z2symmetries == "auto" case
+        # extract problem.hamiltonian and problem.symmetry_sector_locator
+        """Builds a ``TaperedQubitMapper`` from one of the other mappers and from a specific problem.
+        This simplifies the identification of the Pauli operator symmetries and of the symmetry sector
+        in which lies the solution of the problem.
+
+        Args:
+            mapper: ``QubitMapper`` object implementing the mapping of second quantized operators to
+                Pauli operators.
+            problem: A class encoding a problem to be solved.
+
+        Return:
+            A ``TaperedQubitMapper`` with pre-built symmetry specifications.
+        """
+
+        qubit_op, _ = problem.second_q_ops()
+        mapped_op = mapper.map(qubit_op).primitive
+        z2_symmetries = Z2Symmetries.find_z2_symmetries(mapped_op)
+        tapering_values = problem.symmetry_sector_locator(z2_symmetries, mapper)
+        z2_symmetries.tapering_values = tapering_values
+        return TaperedQubitMapper(mapper, z2_symmetries)
+
+    def _map_clifford_single(self, second_q_op: SparseLabelOp) -> SparsePauliOp:
+        mapped_op = self.mapper.map(second_q_op).primitive
+        converted_op = self.z2symmetries.convert_clifford(mapped_op)
+        return converted_op
+
+    def _symmetry_reduce_clifford_single(
+        self, converted_op: SparsePauliOp, check_commutes: bool = True
```

Comment:
```suggestion
        self, converted_op: SparsePauliOp, *, check_commutes: bool = True
```
This makes `check_commutes` a keyword-only argument